tests/core/pthread/create.cpp

```diff
@@ -30,24 +29,20 @@ pthread_t thread[NUM_THREADS];
 
 void CreateThread(int i)
 {
-  static int counter = 1;
   int rc = pthread_create(&thread[i], nullptr, ThreadMain, (void*)i);
   assert(rc == 0);
 }
 
 void mainn() {
   static int main_adds = 0;
-  int worker_adds = sum.load() - main_adds;
-  sum++;
-  main_adds++;
+  int worker_adds = sum++ - main_adds++;
   printf("main iter %d : %d\n", main_adds, worker_adds);
   if (worker_adds == NUM_THREADS * TOTAL) {
     printf("done!\n");
 #ifndef ALLOW_SYNC
-  emscripten_cancel_main_loop();
-#else
-  exit(0);
+    emscripten_cancel_main_loop();
 #endif
+    exit(0);
```

Comment:
Can this go inside the ifndef?

Reply:
It could, but it also requires some other changes to break the while loop, see for example:
```diff
--- a/tests/core/pthread/create.cpp
+++ b/tests/core/pthread/create.cpp
@@ -33,7 +33,7 @@ void CreateThread(int i)
   assert(rc == 0);
 }
 
-void mainn() {
+int mainn() {
   static int main_adds = 0;
   int worker_adds = sum++ - main_adds++;
   printf("main iter %d : %d\n", main_adds, worker_adds);
@@ -41,9 +41,11 @@ void mainn() {
     printf("done!\n");
 #ifndef ALLOW_SYNC
     emscripten_cancel_main_loop();
-#endif
     exit(0);
+#endif
+    return 1;
   }
+  return 0;
 }
 
 int main() {
@@ -55,9 +57,9 @@ int main() {
   // if we don't allow sync pthread creation, the event loop must be reached for
   // the worker to start up.
 #ifndef ALLOW_SYNC
-  emscripten_set_main_loop(mainn, 0, 0);
+  emscripten_set_main_loop((em_callback_func)mainn, 0, 0);
 #else
-  while (1) mainn();
+  while (mainn() == 0) {};
 #endif
   return 0;
 }
```
I thought just calling `exit(0)` might be easier.

Reply:
I see yes,  calling exit() makes the most sense then.